src/miner_util.erl

```diff
@@ -104,16 +104,30 @@ metadata_fun() ->
         Map = blockchain_worker:signed_metadata_fun(),
         case application:get_env(miner, mode, gateway) of
             validator ->
+                RPC =
+                    case lists:filter(fun libp2p_transport_tcp:is_public/1,
+                                      libp2p_swarm:listen_addrs(blockchain_swarm)) of
+                        [] -> #{};
+                        [First | _] ->
+                            IP = lists:nth(2, string:tokens(First, "/")),
+                            {ok, Port} = application:get_env(miner, jsonrpc_port),
```

Comment:
```suggestion
                            Port = application:get_env(miner, grpc_port, 8080),
```
not sure about the name of the port since it's the server listening port and not a client connection port; i think making it configurable was important though to allow the validator hosters to run on different ports behind the same ip/dns name